sentry/tests/test_basic.rs

```diff
@@ -173,3 +174,55 @@ fn test_attached_stacktrace() {
         .flat_map(|ev| ev.threads.into_iter().filter_map(|thrd| thrd.stacktrace));
     assert_eq!(stacktraces.count(), 3);
 }
+
+#[test]
+fn test_attachment_sent_from_scope() {
+    struct TestTransport(Arc<AtomicUsize>);
+
+    impl sentry::Transport for TestTransport {
+        fn send_envelope(&self, envelope: sentry::Envelope) {
```

Comment:
This whole machinery is a bit over-engineered. You have [`sentry::test::with_captured_envelopes`](https://docs.rs/sentry/latest/sentry/test/fn.with_captured_envelopes.html) that does all this for you ;-) No need to define a transport or construct a client/hub.

Reply:
Ah, I think that test was really about testing the transport factory. All the other tests in that file use the appropriate helpers otherwise.